### PR TITLE
Report the PostGIS, geometry native and live migration unsupported types in analyze and assess reports

### PR DESCRIPTION
**Pull request by priyanshi-yb**
Description - 

1. Changed the Assessment query for the datatype of columns to give the UDT type name properly.
2. Added support in the analyze-schema to report the unsupported datatypes completely by db and voyager during Live migration.
https://yugabyte.atlassian.net/browse/DB-12531
https://yugabyte.atlassian.net/browse/DB-12599
3. Added automation tests for these cases in both.
4. Updated expected files of `osm`, `omnibus`,  and `adventureworks` tests.
5. Docs link are not live yet, they are in PR but more or less should be the same. If there will be any change after this PR will fix in follow up.

Jenkins -
pg complex schemas: https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/3624
long-running complex schemas: https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/3629/

### CODE REVIEW

File: migtests/tests/analyze-schema/expected_issues.json [hunk 1]
```diff
@@ -800,7 +800,7 @@
             "Reason": "Unsupported datatype - xml on column - data",
             "SqlStatement": "CREATE TABLE test_xml_type(id int, data xml);",
             "DocsLink": "https://docs.yugabyte.com/preview/yugabyte-voyager/known-issues/postgresql/#data-ingestion-on-xml-data-type-is-not-supported",
-            "Suggestion": "Data ingestion is not supported for this type in YugabyteDB so handle this type in different way. Refer link for more details - \u003cLINK DOC\u003e",
+            "Suggestion": "Data ingestion is not supported for this type in YugabyteDB so handle this type in different way. Refer link for more details.",
```
Comment (makalaaneesh) [inline migtests/tests/analyze-schema/expected_issues.json:803]: even offline is not supported?

File: yb-voyager/cmd/analyzeSchema.go [hunk 2]
```diff
@@ -744,22 +745,42 @@ func reportXMLAndXIDDatatype(createTableNode *pg_query.Node_CreateStmt, sqlStmtI
 		*/
 		if column.GetColumnDef() != nil {
 			typeName := ""
-			if len(column.GetColumnDef().GetTypeName().GetNames()) > 0 {
-				typeName = column.GetColumnDef().GetTypeName().GetNames()[0].GetString_().Sval // 0th index as for these non-native types pg_catalog won't be present on first location
+			typeNames := column.GetColumnDef().GetTypeName().GetNames()
+			if len(typeNames) > 0 {
+				typeName = column.GetColumnDef().GetTypeName().GetNames()[len(typeNames)-1].GetString_().Sval // 0th index as for these non-native types pg_catalog won't be present on first location
```
Comment (makalaaneesh) [inline yb-voyager/cmd/analyzeSchema.go:750]: comment is probably not valid anymore? we are pickking up the last name in typenames.

Do add a comment explaining why we're picking the last

File: yb-voyager/src/srcdb/data/gather-assessment-metadata/postgresql/table-columns-data-types.psql [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
 SELECT
-    table_schema AS schema_name,
-    table_name,
-    column_name,
-    data_type
+    c.table_schema AS schema_name,
+    c.table_name,
+    c.column_name,
+    CASE
+        WHEN c.data_type = 'USER-DEFINED' THEN c.udt_name
```
Comment (makalaaneesh) [inline yb-voyager/src/srcdb/data/gather-assessment-metadata/postgresql/table-columns-data-types.psql:8]: I'm confused about how this fix is relevant here? 
It looks like we are just getting the type name from the SQL statement parse tree, checking if it is part of the list of PostgresUnsupportedDataTypes, that's it. RIght?

Comment (priyanshi-yb) [reply yb-voyager/src/srcdb/data/gather-assessment-metadata/postgresql/table-columns-data-types.psql:8]: This is relevant for reporting these postGis unsupported datatypes like `GEOMETRY / GEOGRAPHY` in Assessment report, as these are user-defined type from extension so it was earlier getting stored as `USER-DEFINED` and we were not able to report these there. So I have modified the query to return the exact udt type name in the data.

File: migtests/tests/analyze-schema/expected_issues.json (hunk 1 above)
Comment (priyanshi-yb) [reply migtests/tests/analyze-schema/expected_issues.json:803]: yeah, YBDB doesn't support inserts on this type.

File: yb-voyager/cmd/analyzeSchema.go (hunk 2 above)
Comment (priyanshi-yb) [reply yb-voyager/cmd/analyzeSchema.go:750]: right, will fix

**Review by makalaaneesh — APPROVED** [review]
LGTM. Just update that comment before merging.

**Pull request merged**